kahuna/public/js/edits/archiver.js

```diff
@@ -1,41 +1,53 @@
 import angular from 'angular';
 import template from './archiver.html!text';
-import '../services/archive';
 
-export var archiver = angular.module('kahuna.edits.archiver', [
-    'kahuna.services.archive'
-]);
+import imageStream from '../image/stream';
 
-archiver.controller('ArchiverCtrl', ['$scope', '$window', 'archiveService', 'onValChange',
-                    function($scope, $window, archiveService, onValChange) {
+import '../archiver/service';
+
+export var archiver = angular.module('kahuna.edits.archiver', ['gr.archiver.service']);
+
+archiver.controller('ArchiverCtrl', ['$window', 'archiverService',
+                    function($window, archiverService) {
 
     var ctrl = this;
 
-    ctrl.toggleArchived = toggleArchived;
-    ctrl.isArchived = ctrl.image.data.userMetadata.data.archived.data;
-    ctrl.archiving = false;
+    const imageStream$ = imageStream([ctrl.image]).getObservable();
+    const service = archiverService(imageStream$);
+
+    ctrl.saving = false;
+    ctrl.isArchived = () => ctrl.image.data.userMetadata.data.archived.data;
 
-    $scope.$watch(() => ctrl.image.data.userMetadata.data.archived.data, onValChange(newState => {
-        ctrl.isArchived = newState;
-    }));
+    ctrl.archive = () => {
+        ctrl.saving = true;
+        service.
+            archive().
+            catch(error).
+            finally(saved);
```

Comment:
Detail, but I find it easier to read if the same locality of code is responsible for setting/clearing the saving flag. Either by mutating it here both times, or having two helper functions defined near each other that set and reset it.